### PR TITLE
Add useful metrics for ClusterPools

### PR DESCRIPTION
**Pull request by 2uasimojo**
[HIVE-1677](https://issues.redhat.com/browse/HIVE-1677)

### CODE REVIEW

File: pkg/controller/clusterpool/metrics.go [hunk 1]
```diff
@@ -7,6 +7,40 @@ import (
...
+	metricClusterDeploymentsDeleting = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "hive_clusterpool_clusterdeployments_deleting",
+		Help: "The number of ClusterDeployments marked for or actively deprovisioning and deleting. Contributes to MaxConcurrent.",
+	}, []string{"clusterpool_namespace", "clusterpool_name"})
+	metricClusterDeploymentsInstalling = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "hive_clusterpool_clusterdeployments_installing",
+		Help: "The number of ClusterDeployments being added to the pool, in the process of installing. Contributes to Size, MaxSize, and MaxConcurrent.",
+	}, []string{"clusterpool_namespace", "clusterpool_name"})
+	// This is always assignable+installing. But it's the number we strive to keep equal to the pool size.
+	// Is it useful to include?
```
Comment (suhanime) [inline pkg/controller/clusterpool/metrics.go:27]: I think that's a simple prom query so I vote for scraping it

Comment (2uasimojo) [reply pkg/controller/clusterpool/metrics.go:27]: But OTOH, the consumer of these metrics would have to understand that internal business logic. Including knowing that it should include "standby" once that's implemented. I'm leaning towards keeping it :)

**Pull request merged**